inc/az_ulib_ustream_forward_base.h

```diff
@@ -0,0 +1,374 @@
+ */
+typedef struct az_ulib_ustream_forward_tag az_ulib_ustream_forward;
+
+/**
+ * @brief Signature of the function to be invoked by the `ustream_forward->flush` operation when the 
+ *        `const az_span* const` buffer has been created.
+ * 
+ * @param[in]   buffer                  The `const az_span* const` buffer to be handled by the 
+ *                                      implementation of this callback.
+ * @param[in]   push_callback_context   The #az_context* contract held between the owner of this
+ *                                      callback and the `ustream_forward->flush` operation.
+ */
+typedef void (*az_ulib_flush_callback)(const az_span* const buffer, az_context* push_callback_context);
+
+/**
+ * @brief   vTable with the ustream_forward APIs.
+ *
+ *  Any module that exposes the ustream_forward shall implement the functions on this vTable.
+ *
+ *  Any code that will use an exposed ustream_forward shall call the APIs using the `az_ulib_ustream_forward_...`
+ *      inline functions.
+ */
+typedef struct az_ulib_ustream_forward_interface_tag
+{
+  /** Concrete `flush` implementation. */
+  az_result (*flush)(
+      az_ulib_ustream_forward* ustream_forward_instance, 
+      az_ulib_flush_callback push_callback, 
+      az_context* push_callback_context);
+
+  /** Concrete `read` implementation. */
+  az_result (*read)(
+      az_ulib_ustream_forward* ustream_forward_instance,
+      uint8_t* const buffer,
+      size_t buffer_length,
+      size_t* const size);
+
+  /** Concrete `get_remaining_size` implementation. */
+  az_result (*get_remaining_size)(
+        az_ulib_ustream_forward* ustream_forward_instance, 
+        size_t* const size);
+
+  /** Concrete `dispose` implementation. */
+  az_result (*dispose)(
+        az_ulib_ustream_forward* ustream_forward_instance);
+} az_ulib_ustream_forward_interface;
+
+/**
+ * @brief   Signature of the function to release memory passed to the ustream_forward
+ *
+ * @param[in]   release_pointer       void pointer to memory that needs to be free'd
+ *
+ */
+typedef void (*az_ulib_release_callback)(void* release_pointer);
+
+/**
+ * @brief   Pointer to the data from which to read
+ *
+ * void pointer to memory where the data is located or any needed controls to access the data.
+ * The content of the memory to which this points is up to the ustream_forward implementation.
+ *
+ */
+typedef void* az_ulib_ustream_forward_data;
+
+/**
+ * @brief   Structure for data control block
+ *
+ * For any given ustream_forward that is created, one control block is created and initialized.
+ *
+ * @note    This structure should be viewed and used as internal to the implementation of the
+ *          ustream_forward. Users should therefore not act on it directly and only allocate the memory
+ *          necessary for it to be passed to the ustream_forward.
+ *
+ */
+typedef struct az_ulib_ustream_forward_data_cb_tag
```

Comment:
>az_ulib_ustream_forward_data_cb_tag [](http://example.com/codeflow?start=15&length=35)

Please encapsulate the content of this structure in `_internal` struct. We need to fix it in the uStream as well.